packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,482 @@
+import { CommittedTransactionInfo, GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  SimpleTransactionBuilder,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeOption,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, assetAmount, assetToBase, baseAmount } from '@xchainjs/xchain-util'
+import axios from 'axios'
+import {
+  AssetXRD,
+  KeyType,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STATE_ENTITY_DETAILS_PATH,
+  STOKENET_GATEWAY_URL,
+  TRANSACTION_COMMITTED_DETAILS_PATH,
+  TRANSACTION_CONSTRUCTION_PATH,
+  TRANSACTION_PREVIEW_PATH,
+  TRANSACTION_SUBMIT_PATH,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+} from './const'
+import { EntityDetailsResponse, Transaction } from './types/radix'
+
+/**
+ * Custom Radix client
+ */
+
+class Client extends BaseXChainClient {
+  privateKey: PrivateKey
```

Comment:
There is an bug in this. You're storing the private key here and the phrase is stored on the base class. If a client calls `radixClient.setPhrase` it updates the seed phrase but doesn't update the private key.

You need to implement `setPhrase`, have it update the private key, and also have it call `super.setPhrase`. Alternatively, do not store the private key and just perform the HD Derivation each time the account address is requested.

I think that it is probably better to store an `HDNode` here so that later methods that require a `walletIndex` become simple to handle. The evm-client for example does this.

Reply:
this should be fixed with the current implementation